src/r2mm/installing/profile_installers/ModLoaderVariantRecord.ts

```diff
@@ -1,217 +1,61 @@
-    WizardWithAGun: MODLOADER_PACKAGES,
-    Sunkenland: MODLOADER_PACKAGES,
-    Atomicrops: MODLOADER_PACKAGES,
-    Erenshor: MODLOADER_PACKAGES,
-    LastTrainOuttaWormtown: MODLOADER_PACKAGES,
-    Dredge: MODLOADER_PACKAGES,
-    CitiesSkylines2: MODLOADER_PACKAGES,
-    LethalCompany: MODLOADER_PACKAGES,
-    MeepleStation: MODLOADER_PACKAGES,
-    VoidCrew: MODLOADER_PACKAGES,
-    Sailwind: MODLOADER_PACKAGES,
-    VotV: MODLOADER_PACKAGES,
-    Palworld: MODLOADER_PACKAGES,
-    Plasma: MODLOADER_PACKAGES,
-    ContentWarning: MODLOADER_PACKAGES,
-    Balatro: MODLOADER_PACKAGES,
-    BoplBattle: MODLOADER_PACKAGES,
-    Vertigo2: MODLOADER_PACKAGES,
-    AgainstTheStorm: MODLOADER_PACKAGES,
-    Lycans: MODLOADER_PACKAGES,
-    CastleStory: MODLOADER_PACKAGES,
-    Panicore: MODLOADER_PACKAGES,
-    RiskofRainReturns: MODLOADER_PACKAGES,
-    Magicraft: MODLOADER_PACKAGES,
-    AnotherCrabsTreasure: MODLOADER_PACKAGES,
-    GladioMori: MODLOADER_PACKAGES,
-    SlipstreamRogueSpace: MODLOADER_PACKAGES,
-    BacktotheDawn: MODLOADER_PACKAGES,
-    BelowTheStone: MODLOADER_PACKAGES,
-    Gloomwood: MODLOADER_PACKAGES,
-    AmongUs: MODLOADER_PACKAGES,
-    BetrayalBeach: MODLOADER_PACKAGES,
-    ArcusChroma: MODLOADER_PACKAGES,
-    AleAndTaleTavern: MODLOADER_PACKAGES,
-    DeepRockGalacticSurvivor: MODLOADER_PACKAGES,
-    ScrewDrivers: MODLOADER_PACKAGES,
-    NineSols: MODLOADER_PACKAGES,
-    GoodbyeVolcanoHigh: MODLOADER_PACKAGES,
-    SupermarketTogether: MODLOADER_PACKAGES,
-    HadesII: MODLOADER_PACKAGES,
-    Shapez2: MODLOADER_PACKAGES,
-    PaqueretteDownTheBunburrows: MODLOADER_PACKAGES,
-    HardTime3: MODLOADER_PACKAGES,
-    TankTeam: MODLOADER_PACKAGES,
-    Distance: MODLOADER_PACKAGES,
-    FiveNightsAtFreddysIntoThePit: MODLOADER_PACKAGES,
-    GoreBox: MODLOADER_PACKAGES,
-    TCGCardShopSimulator: MODLOADER_PACKAGES,
-    OldMarketSimulator: MODLOADER_PACKAGES,
-    Subterranauts: MODLOADER_PACKAGES,
-    SULFUR: MODLOADER_PACKAGES,
-    WEBFISHING: MODLOADER_PACKAGES,
-};
-// Exported separately from the definition in order to preserve the key names in the type definition.
-// Otherwise this would become [key: string] and we couldn't use the game names for type hinting elsewhere.
-// Casting is done here to ensure the values are ModLoaderPackageMapping[]
-export type GAME_NAME = keyof typeof VARIANTS;
-export const MOD_LOADER_VARIANTS: {[key in GAME_NAME]: ModLoaderPackageMapping[]} = VARIANTS;
+const OVERRIDES: { [key: string]: [ModLoaderPackageMapping] } = {
+    BONEWORKS: [
+        new ModLoaderPackageMapping(
+            'LavaGang-MelonLoader',
+            '',
+            PackageLoader.MELON_LOADER,
+            new VersionNumber('0.5.4'),
+        ),
+    ],
+    BONELAB: [
+        new ModLoaderPackageMapping(
+            'LavaGang-MelonLoader',
+            '',
+            PackageLoader.MELON_LOADER,
+        ),
+    ],
+}
+
+// TODO:
+// - Overrides for BONELAB and BONEWORKS.
+// - Better type?
```

Comment:
At least for the old way of adding games the stronger typing was valuable as typos were immediately detected. I'm not sure if that's still true with the new approach?

Reply:
Migrating to the schema means we lose strong typing for game names. I think that we can do a better job of integrating this weird "overrides" system into the schema, but I wasn't sure if this was even still necessary.